modules/dbfs/filestorage.go

```diff
@@ -76,7 +79,110 @@ func (di *DatabaseImpl) FileMove(startRelpath string, startFilename string, endR
 	return err
 }
 
-func calculateFilePath(relpath string, filename string, projectID int64) (string, error) {
+// returns the swap file contents and any error
+func (di *DatabaseImpl) makeSwp(relpath string, filename string, projectID int64) ([]byte, error) {
```

Comment:
'swp' is the standard abbreviation for 'swap', you'll find that even linux uses it to name it's swap files.